azurerm/resource_arm_storage_management_policy.go

```diff
@@ -0,0 +1,389 @@
+package azurerm
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/terraform/helper/validation"
+
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2019-04-01/storage"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmStorageManagementPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create:        resourceArmStorageManagementPolicyCreateOrUpdate,
+		Read:          resourceArmStorageManagementPolicyRead,
+		Update:        resourceArmStorageManagementPolicyCreateOrUpdate,
+		Delete:        resourceArmStorageManagementPolicyDelete,
+		SchemaVersion: 1,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"storage_account_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
```

Comment:
We should add validation here
```suggestion
				ForceNew: true,
				ValidateFunc: azure.ValidateResourceID,
```